src/libraries/drand/_index.md

```diff
@@ -0,0 +1,120 @@
+- `PreviousSignature`   -- the threshold BLS signature from the previous drand round.
+- `Randomness`          --  the SHA256 hash of `Signature`, to be used as the random value for this round.
+- `Round`               -- the index of Randomness in the sequence of all random values produced by this drand network.
+
+Specifically, the message signed is the concatenation of the round number treated as a uint64 and the previous signature. At the moment, drand uses BLS signatures on the BN256 curves and the signature is made over G1.
+
+### Polling the drand network
+
+Filecoin nodes can make use of [drand endpoints](https://github.com/drand/drand/blob/master/core/client_public.go) in working with a drand beacon.
+
+To start, a Filecoin node must store a set of drand peer servers it will connect to to poll for values and the shared public key it expects them to have (TODO: explain how this list/key is obtained in below section). Polling is done using gRPCs (and protobufs).
+
+On initialization, the Filecoin node can call the `Group` endpoint in order to obtain the beacon's [group file](https://github.com/drand/drand/blob/57a6056a24d4ebaa27a44852636807364624b9fc/key/group.go). Using this, the Filecoin node should:
+
+- cache the beacon's `Period`                           -- the period of time between each drand randomness generation
+- cache the beacon's `GenesisTime`                      -- at which the first round in the drand randomness chain is created
+- verify that the beacon's `PublicKey` is appropriate   -- ie that the filecoin node connected to the right drand beacon
+
+Thereafter, the Filecoin client can call drand's endpoints:
+
+- `LastPublic` to get the latest randomness value produced by the beacon
+- `Public` to get the randoomness value produced by the beacon at a given index
+
+{{<label drand>}}
+### Using drand in Filecoin
+
+Drand is used as a randomness beacon for leader election in Filecoin. You can read more about that in {{<sref leader_election>}}. See drand used in the Filecoin lotus implementation [here](https://github.com/filecoin-project/lotus/blob/master/chain/beacon/drand/drand.go).
+
+While drand returns multiple values with every call to the beacon (see above), Filecoin blocks need only store a subset of these in order to track a full drand chain. This information can then be mixed with on-chain data for use in Filecoin. See {{<sref randomness>}} for more.
+
+#### Verifying an incoming drand value
+
+Upon receiving a new drand randomness value from a beacon, a Filecoin node should immediately verify its validity. That is, it should verify:
+
+- that the `Signature` field is verified by the beacon's `PublicKey` as the beacon's signature of `SHA256(PreviousSignature || Round)`.
+- that the `Randomness` field is `SHA256(Signature)`.
+
+See [drand](https://github.com/drand/drand/blob/master/core/client_public.go#L93) for an example.
```

Comment:
I would link to the more core library https://github.com/drand/drand/blob/master/beacon/beacon.go#L63